### PR TITLE
Make the URI-label in the properties dialog a button showing the folder

### PR DESCRIPTION
**Pull request by raveit65**
UI changes are from myself.
Please test.

### CODE REVIEW

File: src/eom-properties-dialog.c [hunk 1]
```diff
@@ -165,15 +166,21 @@ pd_update_general_tab (EomPropertiesDialog *prop_dlg,
...
-	uri_str = eom_image_get_uri_for_display (image);
-	dir_str = g_path_get_dirname (uri_str);
-	gtk_label_set_text (GTK_LABEL (prop_dlg->priv->location_label),
-			    dir_str);
+	parent_file = g_file_get_parent (file);
+	if (parent_file == NULL) {
+		/* file is root directory itself */
+		parent_file = g_object_ref (file);
+	}
+	dir_str = g_file_get_basename (parent_file);
```
Comment (vkareh) [inline src/eom-properties-dialog.c]: Could we use `g_file_get_path` here? If I have an image in `/home/vkareh/Pictures/2018/07/16/foo.jpg`, the button displays just "16", which is not very useful. Using `g_file_get_path` gives you the full path as the label.

File: src/eom-properties-dialog.c [hunk 2]
```diff
@@ -401,6 +408,23 @@ pd_close_button_clicked_cb (GtkButton *button,
...
+	guint32 timestamp;
+
+
+	if (!priv->folder_button_uri)
+		return;
+	
+	screen = gtk_widget_get_screen (GTK_WIDGET (button));
+	timestamp = gtk_get_current_event_time ();
+
+	gtk_show_uri (screen, priv->folder_button_uri, timestamp, NULL);
```
Comment (vkareh) [inline src/eom-properties-dialog.c]: This gives me a deprecated warning. We should use `gtk_show_uri_on_window` instead.

File: src/eom-properties-dialog.c (hunk 1 above)
Comment (raveit65) [reply src/eom-properties-dialog.c]: So , `dir_str = g_file_get_path (parent_file);` ?

File: src/eom-properties-dialog.c (hunk 2 above)
Comment (raveit65) [reply src/eom-properties-dialog.c]: I will fix that with fixing another gtk_show_uri waring in code. I have local prepared a back port of this one https://gitlab.gnome.org/GNOME/eog/commit/77f16df

**Comment by raveit65** [comment]
Using `dir_str = g_file_get_path (parent_file);` work good, but how long can a linux file path be?
eom-details-button
I am a bit in worry to increase the windows width.

**Comment by vkareh** [comment]
Oh, i know! Keep the current basepath as you made it, but make the tooltip show the full path, would that work for you?

**Comment by raveit65** [comment]
Looks like we need some scaling of the image for HIDPI.

**Comment by vkareh** [comment]
@raveit65 - I updated the sidebar to also show the full path in the tooltip: https://github.com/mate-desktop/eom/pull/178/commits/42135c7122d840ca0aeaeeb843ed99e60d7a55aa

**Comment by raveit65** [comment]
Good idea with the sidebar, i will test it and merge the PR if it works for me.

**Comment by raveit65** [comment]
I fixed a little typo in last commit and rebased PR.
```
eom-metadata-sidebar.c: In function 'eom_metadata_sidebar_update_general_section':
/usr/include/glib-2.0/gobject/gtype.h:2224:6: warning: passing argument 1 of 'gtk_widget_set_tooltip_text' from incompatible pointer type [-Wincompatible-pointer-types]
     ((ct*) g_type_check_instance_cast ((GTypeInstance*) ip, gt))
/usr/include/glib-2.0/gobject/gtype.h:482:66: note: in expansion of macro '_G_TYPE_CIC'
 #define G_TYPE_CHECK_INSTANCE_CAST(instance, g_type, c_type)    (_G_TYPE_CIC ((instance), (g_type), c_type))
                                                                  ^~~~~~~~~~~
/usr/include/gtk-3.0/gtk/gtkbutton.h:40:42: note: in expansion of macro 'G_TYPE_CHECK_INSTANCE_CAST'
 #define GTK_BUTTON(obj)                 (G_TYPE_CHECK_INSTANCE_CAST ((obj), GTK_TYPE_BUTTON, GtkButton))
                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~
eom-metadata-sidebar.c:271:31: note: in expansion of macro 'GTK_BUTTON'
  gtk_widget_set_tooltip_text (GTK_BUTTON (priv->folder_button), str);
                               ^~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtkapplication.h:27,
                 from /usr/include/gtk-3.0/gtk/gtkwindow.h:33,
                 from /usr/include/gtk-3.0/gtk/gtkdialog.h:32,
                 from /usr/include/gtk-3.0/gtk/gtkaboutdialog.h:30,
                 from /usr/include/gtk-3.0/gtk/gtk.h:31,
                 from eom-metadata-sidebar.c:33:
/usr/include/gtk-3.0/gtk/gtkwidget.h:1275:59: note: expected 'GtkWidget *' {aka 'struct _GtkWidget *'} but argument is of type 'GtkButton *' {aka 'struct _GtkButton *'}
 void       gtk_widget_set_tooltip_text      (GtkWidget   *widget,
                                              ~~~~~~~~~~~~~^~~~~~
```

**Comment by raveit65** [comment]
Tooltip is fine in sidebar button, i will merge it.

File: src/eom-properties-dialog.c (hunk 2 above)
Comment (raveit65) [reply src/eom-properties-dialog.c]: @vkareh 
`gtk_show_uri_on_window` needs a GtkWindow as parent, but eom-properties-dialog is a GtkDialog window.
In eog the dialog is ported to a GtkApplication window after porting away from GtkAction.
So the solution from them doesn't work.
https://gitlab.gnome.org/GNOME/eog/commit/ac4f44b56554459c2fee7b849635ce7769230cfc
Any Idea?

**Pull request merged**